internal/terraform/context_plan.go

```diff
@@ -513,6 +513,33 @@ func (c *Context) postPlanValidateMoves(config *configs.Config, stmts []refactor
 	return refactoring.ValidateMoves(stmts, config, allInsts)
 }
 
+// All import target addresses with a key must already exist in config.
+// When we are able to generate config for expanded resources, this rule can be
+// relaxed.
+func (c *Context) postPlanValidateImports(config *configs.Config, importTargets []*ImportTarget, allInst instances.Set) tfdiags.Diagnostics {
```

Comment:
I realize it's probably too late in the project to be worth this sort of refactoring right now, but FWIW for `moved` blocks we put similar functionality in the `internal/refactoring` package in an attempt to keep the `terraform` package's overall sprawl contained, and maybe the validation rules and such for import would be at home in there too, so the `terraform` package can focus just on the mechanics of doing the import while delegating the "rules" for import to another place that we can have independent unit tests for.

(I don't mean this as blocking feedback. No problem with having this here to start, but might be nice to consider for some refactoring later once this new feature has settled in a bit.)

Reply:
Yes, definitely agree. We can also decide whether we want to reuse any of the moved address types (`addrs.MoveEndpointInModule` etc) for import target addresses when we give them some more sophisticated handling.

CJ has a couple more move validation PRs coming - after they're all merged I'll aim to refactor the validation into `refactoring`.